network/src/network.rs

```diff
@@ -125,13 +126,14 @@ impl NetworkState {
             peer_store,
             config,
             peer_registry: RwLock::new(peer_registry),
-            dialing_addrs: RwLock::new(FnvHashMap::default()),
-            listened_addresses: RwLock::new(listened_addresses),
-            original_listened_addresses: RwLock::new(Vec::new()),
-            disconnecting_sessions: RwLock::new(FnvHashSet::default()),
+            dialing_addrs: RwLock::new(HashMap::default()),
+            listened_addrs: RwLock::new(listened_addrs),
+            original_listened_addrs: RwLock::new(Vec::new()),
```

Comment:
We don't need `original_listened_addrs` anymore. `NetworkService::start` can also use the vote interface and give the binded addr a very high score.

Reply:
We use `u8` to record a score, if we remove `original_listened_addrs`, once node up long enough, we can't distinguish `original_listened_addrs` from `listened_addrs`.

IMO we should keep the distinguish, and keep the `original_listened_addrs`, it not slow down the code, and not confuse the code(if so we can chose a better name, the concept is simple).

In the future, we may introduce another vote or expired mechanism to `listened_addr` but not `original_listened_addrs`. Keep the simple concept will not complex our code IMO.

Reply:
Overload a single sctructure brings more confuse and complexity IMO.

Reply:
I think it's better to rename these fields:

- `listened_addrs` to `observed_addrs`, `public_addrs` or `external_addrs`. Also, we use external in code, public in config file, maybe it's better to keep consistent on the terminology.
- `original_listened_addrs` to just `listened_addrs`.

Reply:
I'd like do the following changes:

* `listened_addrs` to `external_addrs`.
* `original_listened_addrs` to `listened_addrs`

`public_addrs` is used in the case that ckb can't find external IP through protocols, for example, the node is behandling a NAT or a reverse proxy, user can manually set public IP.

* `listened_addrs` - user config listened addresses.
* `public_addrs` - user config public addrs.
* `external_addrs` - addresses that discovered through identify + `public_addrs`